File: packages/rook_sdk/android/src/main/kotlin/io/tryrook/rook_sdk/mapper/HealthPermissionMappers.kt
package io.tryrook.rook_sdk.mapper

import com.rookmotion.rook.sdk.domain.enums.HealthPermission
import io.tryrook.rook_sdk.exception.UnknownPermissionException
import io.tryrook.rook_sdk.proto.HealthPermissionProto

fun HealthPermissionProto.toDomain(): HealthPermission {
    return when (this) {
        HealthPermissionProto.SLEEP -> HealthPermission.SLEEP
        HealthPermissionProto.PHYSICAL -> HealthPermission.PHYSICAL
        HealthPermissionProto.BODY -> HealthPermission.BODY
        HealthPermissionProto.ALL -> HealthPermission.ALL
        else -> throw UnknownPermissionException()
    }
}

File: packages/rook_sdk/lib/src/data/mapper/health_permission_mappers.dart
import 'package:rook_sdk/src/data/proto/protos.pbenum.dart';
import 'package:rook_sdk/src/domain/enums/health_permission.dart';

extension HealthPermissionMappers on HealthPermission {
  HealthPermissionProto toProto() {
    return switch (this) {
      HealthPermission.sleep => HealthPermissionProto.SLEEP,
      HealthPermission.body => HealthPermissionProto.BODY,
      HealthPermission.physical => HealthPermissionProto.PHYSICAL,
      _ => HealthPermissionProto.ALL,
    };
  }
}
